recipe/meta.yaml

```diff
@@ -7,13 +7,14 @@ source:
   sha256: 5db25d4fd138013b563f9a3d1d87f7de7df1dac014fd4cccdfbb3435a5cff761
 
 build:
-  number: 0
+  number: 1
   noarch: generic
 
 requirements:
   run_constrained:
     {% if version is defined %}                # Safe-guard for early parsing
     - __cuda >={{ version.split(".", 1)[0] }}  # Major version is lower bound
+    - cudatoolkit {{ version }}                # Align `cudatoolkit` version
```

Comment:
Should this only constrain for versions less than 12? We don’t have a cudatoolkit 12.0 and I am not sure if we will have one.

Reply:
Ah. I see the flaw in my logic. Maybe the constraint should match the cudatoolkit version for CUDA <12 and be constrained to <0.0.0a0 (no permitted cudatoolkit) for CUDA 12 and above? I wanted to avoid defining a constraint on a package version that doesn’t exist, but it could be argued that <0.0.0a0 doesn’t exist either. Maybe my argument is not a good one, so I can retract this idea if others prefer constraining on cudatoolkit=12.0 for CUDA 12 despite it not existing.

Reply:
The outcome largely depends on what we plan to do with cudatoolkit vs. cuda-toolkit (the metapackage name on the nvidia channel). cc: @adibbley @jakirkham for input.

Reply:
Think that is the strength of this approach as opposed to adding `cuda-version` to `cudatoolkit`. The latter requires a repodata patch to `cudatoolkit`, which not all channels would be able to add. Whereas `cuda-version` (currently) is only packaged here (and not for very long), so adopting the constraint in `cuda-version` properly constrains `cudatoolkit`s from other channels

Reply:
Speaking of `cuda-toolkit`, I think we could add it to `run_constrained` here as well, but traditionally CF doesn't depend on or contain packages from any other channels in anyway, so not sure if it'd be OK or not.

Reply:
Think `cuda-toolkit` (from the `conda-forge` perspective) will be handled when we add that package ( https://github.com/conda-forge/staged-recipes/issues/21382 ), which will contain the `cuda-version` dependency (as other CTK packages have)

If we want to do more than that, would suggest we raise an issue to discuss separately

Reply:
Thanks @jakirkham for the ideas here. I think this sounds like a solid plan. Approved.